### PR TITLE
Json schema Generation Guide

### PR DESCRIPTION
**Pull request by sdelamo**
To generate the guide locally: `./gradlew micronautJsonSchemaBuild`

serve it locally: 

`jwebserver -d /Users/sdelamo/github/micronaut-projects/micronaut-guides/build/dist`

and you can open the guide: 

http://localhost:8000/micronaut-json-schema-gradle-java.html

@AndreaLaGrotteria you can see how Micronaut JSON Schema works

### CODE REVIEW

File: guides/micronaut-json-schema/java/src/main/java/example/micronaut/Product.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.jsonschema.JsonSchema;
+import jakarta.validation.constraints.*;
+
+import java.util.Set;
+
+@JsonSchema(description = "A product from Acme's catalog") // <1>
+public record Product(
+        @NotNull // <2>
+        @JsonPropertyDescription("The unique identifier for a product") // <3>
```
Comment (graemerocher) [inline guides/micronaut-json-schema/java/src/main/java/example/micronaut/Product.java]: can we also demonstrate the use of javadoc to propagate descriptions? I believe that is possible /cc @andriy-dmytruk

File: guides/micronaut-json-schema/java/src/test/java/example/micronaut/JsonSchemaGeneratedTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@MicronautTest(startApplication = false) // <1>
+class JsonSchemaGeneratedTest {
+
+    @Test
+    void buildGeneratesOpenApi(ResourceLoader resourceLoader) {
```
Comment (graemerocher) [inline guides/micronaut-json-schema/java/src/test/java/example/micronaut/JsonSchemaGeneratedTest.java]: method name seems wrong?

File: guides/micronaut-json-schema/java/src/test/java/example/micronaut/ProductSchemaTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@MicronautTest
+class ProductSchemaTest {
+
+    @Test
+    void testProductSchema(@Client("/")HttpClient httpClient) throws JSONException {
+        BlockingHttpClient client = httpClient.toBlocking();
+        URI uri = UriBuilder.of("/schemas").path("product.schema.json").build();
+        HttpRequest<?> request = HttpRequest.GET(uri);
```
Comment (graemerocher) [inline guides/micronaut-json-schema/java/src/test/java/example/micronaut/ProductSchemaTest.java]: unclear why `UriBuilder` is used here

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: guides/micronaut-json-schema/java/src/test/java/example/micronaut/ProductSchemaTest.java (hunk 3 above)
Comment (sdelamo) [reply guides/micronaut-json-schema/java/src/test/java/example/micronaut/ProductSchemaTest.java]: No particular reason. I like to show the `UriBuilder` API. I have changed it here: https://github.com/micronaut-projects/micronaut-guides/pull/1495/commits/102cf51b8109fc85c7fa101d5d9afecbf8d4627e

File: guides/micronaut-json-schema/java/src/test/java/example/micronaut/JsonSchemaGeneratedTest.java (hunk 2 above)
Comment (sdelamo) [reply guides/micronaut-json-schema/java/src/test/java/example/micronaut/JsonSchemaGeneratedTest.java]: fixed via https://github.com/micronaut-projects/micronaut-guides/pull/1495/commits/73d6845b812fc64c32f525006f56651d23061279

File: guides/micronaut-json-schema/java/src/main/java/example/micronaut/Product.java (hunk 1 above)
Comment (sdelamo) [reply guides/micronaut-json-schema/java/src/main/java/example/micronaut/Product.java]: I tried to use javadoc description originally but it does not work. 

For example using java doc for `productName` causes the test to fail. 

```java
@JsonSchema(description = "A product from Acme's catalog") // <1>
public record Product(
        @NotNull // <2>
        @JsonPropertyDescription("The unique identifier for a product") // <3>
        Long productId,

        /**
         * Name of the product
         */
        @NonNull
        @NotBlank
        String productName,

        @NotNull // <1>
        @JsonPropertyDescription("The price of the product") // <3>
        @Positive // <4>
        Double price,

        @Size(min = 1) // <4>
        Set<String> tags
) {
}
```

Comment (sdelamo) [reply guides/micronaut-json-schema/java/src/main/java/example/micronaut/Product.java]: Using top level javadoc works: 

```java


/**
 *
 * @param productId  The unique identifier for a product // <1>
 * @param productName Name of the product
 * @param price The price of the product
 * @param tags Product tags
 */
@JsonSchema(description = "A product from Acme's catalog") // <2>
public record Product(
        @NotNull // <3>
        Long productId,

        @NonNull
        @NotBlank
        String productName,

        @NotNull // <3>
        @Positive // <4>
        Double price,

        @Size(min = 1) // <4>
        Set<String> tags
) {
}
```

**Review by graemerocher — APPROVED** [review]

**Comment by andriy-dmytruk** [comment]
This is awesome, @sdelamo!

**Pull request merged**